spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/Mqttv5PahoMessageDrivenChannelAdapter.java

```diff
@@ -106,6 +110,30 @@ public Mqttv5PahoMessageDrivenChannelAdapter(MqttConnectionOptions connectionOpt
 		}
 	}
 
+	/**
+	 * Use this constructor when you need to use a single {@link ClientManager}
+	 * (for instance, to reuse an MQTT connection).
+	 * @param clientManager The client manager.
+	 * @param topic The topic(s).
+	 * @since 6.0
+	 */
+	public Mqttv5PahoMessageDrivenChannelAdapter(ClientManager<IMqttAsyncClient, MqttConnectionOptions> clientManager,
+			String... topic) {
+		super(clientManager, topic);
+		this.connectionOptions = clientManager.getConnectionInfo();
+		setClientManagerCallback(new AdapterConnectCallback());
```

Comment:
Right, no reason in this extra API: just do that `clientManager.addCallback(new AdapterConnectCallback());` directly instead of this and that in the `onInit()`!
Or otherwise if we cannot have an instance of that internal class without this outer one...